test/conformance/source/environment.cpp

```diff
@@ -256,44 +256,49 @@ std::string
 KernelsEnvironment::getKernelSourcePath(const std::string &kernel_name,
                                         uint32_t device_index) {
     std::stringstream path;
-    path << instance->getKernelDirectory();
-    // il_postfix = supported_IL(SPIRV-PTX-...) + IL_version + extension(.spv -
-    // .ptx - ....)
+    path << kernel_options.kernel_directory << "/" << kernel_name;
     std::string il_postfix = getSupportedILPostfix(device_index);
 
     if (il_postfix.empty()) {
         error = "failed getting device supported IL";
         return {};
     }
 
-    path << "/" << kernel_name << il_postfix;
+    std::string binary_name;
+    for (const auto &entry : std::filesystem::directory_iterator(path.str())) {
+        auto file_name = entry.path().filename().string();
+        if (file_name.find(il_postfix) != std::string::npos) {
+            binary_name = file_name;
+            break;
+        }
+    }
 
-    uint32_t address_bits;
-    auto device = instance->GetDevices()[device_index];
-    if (uur::GetDeviceAddressBits(device, address_bits)) {
-        error = "failed getting device address bits supported";
+    if (binary_name.empty()) {
+        error = "couldn't find appropriate device binary";
```

Comment:
Wonder if some of these hard-coded error messages could be standardised in one location or added to the UR_RESULT_X enums maybe?

Reply:
alright they're now used properly and visible in test output